From: Mathieu Trudel-Lapierre <mathieu@canonical.com>
Date: Fri, 20 May 2016 14:32:17 +0800
Subject: On Ubuntu, we use premade icons for the indicator VPN case.

The icons are copies of the original with the padlock pre-composited, becuase
indicators can't use pixbufs; we use the suffix -secure.
---
 src/applet.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/applet.c b/src/applet.c
index b566709..af1e3d4 100644
--- a/src/applet.c
+++ b/src/applet.c
@@ -2759,6 +2759,10 @@ applet_update_icon (gpointer user_data)
 		switch (vpn_state) {
 		case NM_VPN_CONNECTION_STATE_ACTIVATED:
 			icon_name = "nm-vpn-active-lock";
+#ifdef WITH_APPINDICATOR
+			if (with_appindicator)
+				icon_name = icon_name_free = g_strdup_printf ("%s-secure", app_indicator_get_icon (applet->app_indicator));
+#endif /* WITH_APPINDICATOR */
 			break;
 		case NM_VPN_CONNECTION_STATE_PREPARE:
 		case NM_VPN_CONNECTION_STATE_NEED_AUTH:
